lib/fizzy/instantiate.cpp

```diff
@@ -470,18 +471,28 @@ std::optional<FuncIdx> find_exported_function(const Module& module, std::string_
     return find_export(module, ExternalKind::Function, name);
 }
 
+ExecuteFunction::ExecuteFunction(Instance& instance, FuncIdx func_idx)
+  : m_function([](std::any host_context, Instance&, const Value* args, int depth) {
+        auto [idx, inst] = std::any_cast<std::pair<FuncIdx, Instance*>>(host_context);
+        return execute(*inst, idx, args, depth);
+    }),
+    m_host_context(std::make_any<std::pair<FuncIdx, Instance*>>(func_idx, &instance))
```

Comment:
If I understand libstd++ code correctly, small object optimization happens only for objects smaller than pointer size, so putting this `pair` into `any` will still do heap allocation, so I guess Pawel's optimization should make sense, I will squash it.
(too bad we don't have benchmarks that import wasm functions)